### DISCUSSION TITLE
How to connect an object to a connection line

### DISCUSSION THREAD

**Discussion by abdullahahmed-dev** [terrastruct/d2#2156] [Q&A]
I am using D2Lang on VSCode with D2 plugin and so far loving it. My only question is that how do I draw a connection between the arrow shaft and a new object. I want to achieve sth like this. 
![Figure 1](https://github.com/user-attachments/assets/bdc8ab6d-224b-487b-a391-affc8a99b9b9)
The code I am using renders and image sth like this
<img width="580" alt="image" src="https://github.com/user-attachments/assets/f480beec-c5d9-495e-a288-a64dfae164f6">
My code:
```d2
grid-rows: 7
grid-columns: 2

a: 110 Records identified\nthrough database searching {
  shape: rectangle
  style.stroke-width: 2
}

b: 18 Duplicate Records Removed {
  shape: rectangle
  style.stroke-width: 2
  style.stroke: black
}

c: 92 Potential Articles Screened {
  shape: rectangle
  style.stroke-width: 2
}

d: 71 Records excluded based on\nTitle & Abstract {
  shape: rectangle
  style.stroke-width: 2
}

e: 21 Full Texts obtained\nfor detailed review {
  shape: rectangle
  style.stroke-width: 2
}

f: 4 Studies excluded due to\nDesign Heterogeneity {
  shape: rectangle
  style.stroke-width: 2
}

g: 3 Studies did not meet\nInclusion Criteria {
  shape: rectangle
  style.stroke-width: 2
}

h: 1 Study was not in English {
  shape: rectangle
  style.stroke-width: 2
}

i: 13 Studies included in\nthe Final Analysis {
  shape: rectangle
  style.stroke-width: 2
}

a -> b -> c
a -> c
c -> d
c -> e
e -> f
e -> g
e -> h
e -> i

style: {
  fill: white
  stroke: black
  font-size: 14
  shadow: false
}
```

If anybody has any ideas on how can I make my D2 Graph make as much closer to the original picture (the color of boxes, the shadow of boxes, the connection, image dimensions to resize it horizontally) I posted please let me know. It wil be highly appreciated. Thanks

**Comment by alixander** [comment]
It's a longstanding todo: https://github.com/terrastruct/d2/issues/423

**Reply by abdullahahmed-dev** [reply]
I get it. It's not possible. Is there any way I can change font size and font family directly through my editor. Like in plantUML I just type `skinparam DefaultFontName Avenir`. Is it possible to do the same in D2 ? Also is there any way to increase dpi of exported image ? Thanks

**Reply by alixander** [reply]
font size yes, font family no. https://d2lang.com/tour/style/#font-size

there is not a way to increase dpi right now
